servicetalk-client-api/src/main/java/io/servicetalk/client/api/ConnectionLimitReachedException.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright © 2022 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.client.api;
+
+import java.net.ConnectException;
+
+/**
+ * Thrown when the number of connections reached their limit for a given resource (i.e. a host)
+ * depending on the context.
+ */
```

Comment:
Consider adding a `@see` tag with a link to `LimitingConnectionFactoryFilter` and enhancing `LimitingConnectionFactoryFilter`'s javadoc with explanation what exception type will be thrown when the limit is reached.